### DISCUSSION TITLE
MQTT Publish events - gestures - not sent to home assistant

### DISCUSSION THREAD

**Discussion by tom-lejeune** [seaky/nspanel_pro_tools_apk#94] [Q&A]
In the Integration setup of MQTT, I have Publish events:
- Device wake up
- Device sleep
- Gesture -  Tap
- Gesture - swipe down
- etc

in home assistant, I see all the events, exept under "touch", I don't gate the state attributes for the Gestures:

![image](https://github.com/seaky/nspanel_pro_tools_apk/assets/2422652/7ed7efcc-f0a4-4944-9129-8f1750e76d32)

Is this just me or is this a bug?
It says inthe manual that it should work:

action:
  - choose:
      - conditions:
          - condition: state
            entity_id: event.nspanel2_touch
            attribute: event_type
            state: swipe_up

thanks.

**Comment by seaky** [comment]
Hi,

It will work only if the screen is off. Are you sure you are trying when the screen is off?

Regards

**Comment by tom-lejeune** [comment]
Hi,

tried it with screen off as wel, status stays "unknown".

not the worst issue, but just wanted to let you know, no matter what I try,
the status of "touch" stays "unknown".

greetings,
Tom

**Reply by feinadam** [reply]
What is the reason why the touch sensor is active only when the screen is off?

**Reply by seaky** [reply]
On Oreo (8.1) you can not captucre touch events, if the application is not in front.
My workaround was to write a low level touch device event processor which interprets kernel level events.
- until Sonoff firmware 1.11 it is possible capture data and my base intention was to handle action without turning the screen on. For example one tap - turns light on, swipe down - close shutter etc using as a versatile switch .
- from 1.11 the kernel was modified and every touch event will wake up the device. Thats the reason why I introduced the touch panel see Oreo 8.1 limitations.
- operating low level touch interpreter all the time could be an issue what I have to investigate longer...
